aws/resource_aws_backup_selection_test.go

```diff
@@ -0,0 +1,243 @@
+					testAccCheckAwsBackupSelectionExists("aws_backup_selection.test", &selection1),
+					resource.TestCheckResourceAttr("aws_backup_selection.test", "resources.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsBackupSelectionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).backupconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_backup_selection" {
+			continue
+		}
+
+		input := &backup.GetBackupSelectionInput{
+			BackupPlanId: aws.String(rs.Primary.Attributes["plan_id"]),
+			SelectionId:  aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetBackupSelection(input)
+
+		if err == nil {
+			if *resp.SelectionId == rs.Primary.ID {
+				return fmt.Errorf("Selection '%s' was not deleted properly", rs.Primary.ID)
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsBackupSelectionExists(name string, selection *backup.GetBackupSelectionOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("not found: %s, %v", name, s.RootModule().Resources)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).backupconn
+
+		input := &backup.GetBackupSelectionInput{
+			BackupPlanId: aws.String(rs.Primary.Attributes["plan_id"]),
+			SelectionId:  aws.String(rs.Primary.ID),
+		}
+
+		output, err := conn.GetBackupSelection(input)
+
+		if err != nil {
+			return err
+		}
+
+		*selection = *output
+
+		return nil
+	}
+}
+
+func testAccCheckAwsBackupSelectionDisappears(selection *backup.GetBackupSelectionOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).backupconn
+
+		input := &backup.DeleteBackupSelectionInput{
+			BackupPlanId: selection.BackupPlanId,
+			SelectionId:  selection.SelectionId,
+		}
+
+		_, err := conn.DeleteBackupSelection(input)
+
+		return err
+	}
+}
+
+func testAccBackupSelectionConfigBase(rInt int) string {
+	return fmt.Sprintf(`
+data "aws_caller_identity" "current" {}
+
+resource "aws_backup_vault" "test" {
+  name = "tf_acc_test_backup_vault_%d"
+}
+
+resource "aws_backup_plan" "test" {
+  name = "tf_acc_test_backup_plan_%d"
+
+  rule {
+    rule_name         = "tf_acc_test_backup_rule_%d"
+    target_vault_name = "${aws_backup_vault.test.name}"
+    schedule          = "cron(0 12 * * ? *)"
+  }
+}
+`, rInt, rInt, rInt)
+}
+
+func testAccBackupSelectionConfigBasic(rInt int) string {
+	return testAccBackupSelectionConfigBase(rInt) + fmt.Sprintf(`
+resource "aws_backup_selection" "test" {
+  plan_id      = "${aws_backup_plan.test.id}"
+
+  name         = "tf_acc_test_backup_selection_%d"
+  iam_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/service-role/AWSBackupDefaultServiceRole"
+
+  selection_tag {
+    type = "STRINGEQUALS"
+    key = "foo"
+    value = "bar"
+  }
+
+  resources = [
+    "arn:aws:ec2:us-east-1:${data.aws_caller_identity.current.account_id}:volume/"
```

Comment:
The acceptance testing runs in `us-west-2` by default:

https://github.com/terraform-providers/terraform-provider-aws/blob/26fea575f2100d68307e063a8b1c590d8de622d6/aws/provider_test.go#L181-L185

```
--- FAIL: TestAccAwsBackupSelection_withResources (13.48s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
        	* aws_backup_selection.test: 1 error occurred:
        	* aws_backup_selection.test: error creating Backup Selection: InvalidParameterValueException: Invalid ARN: arn:aws:elasticfilesystem:us-east-1:--OMITTED--:file-system/. Expected region is us-west-2
        	status code: 400, request id: 8c64a59c-7b8c-46de-8e7e-6fa1a7879b3a

--- FAIL: TestAccAwsBackupSelection_disappears (13.53s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
        	* aws_backup_selection.test: 1 error occurred:
        	* aws_backup_selection.test: error creating Backup Selection: InvalidParameterValueException: Invalid ARN: arn:aws:ec2:us-east-1:--OMITTED--:volume/. Expected region is us-west-2
        	status code: 400, request id: bb483272-0b15-4978-88ab-0a5cf2400302

--- FAIL: TestAccAwsBackupSelection_basic (13.60s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
        	* aws_backup_selection.test: 1 error occurred:
        	* aws_backup_selection.test: error creating Backup Selection: InvalidParameterValueException: Invalid ARN: arn:aws:ec2:us-east-1:--OMITTED--:volume/. Expected region is us-west-2
        	status code: 400, request id: 6e4c8dda-6cc8-4d63-be62-b4b156e4c30b

--- FAIL: TestAccAwsBackupSelection_withTags (13.66s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
        	* aws_backup_selection.test: 1 error occurred:
        	* aws_backup_selection.test: error creating Backup Selection: InvalidParameterValueException: Invalid ARN: arn:aws:ec2:us-east-1:--OMITTED--:volume/. Expected region is us-west-2
        	status code: 400, request id: a1229860-b1f7-4462-9b22-68d46cf62e31
```

We generally prefer using the `aws_partition` and `aws_region` data sources in this case to make the testing region and partition agnostic:

```hcl
data "aws_partition" "current" {}

data "aws_region" "current" {}

// ...
"arn:${data.aws_partition.current.partition}:ec2:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:volume/"
```